homeassistant/components/climate/zha.py

```diff
@@ -0,0 +1,258 @@
+                                      ['local_temp',
+                                       'pi_heating_demand',
+                                       'occupied_heating_setpoint',
+                                       'setpoint_mode_manuf_specific'])
+        self.last_request = time.time()
+        if not result:
+            _LOGGER.warning("No result received")
+            return
+
+        try:
+            self._heating_power = int(result['pi_heating_demand'])
+        except (TypeError, KeyError):
+            _LOGGER.warning("Can not get heating power value")
+            self._heating_power = None
+
+        raw_mode = result.get('setpoint_mode_manuf_specific')
+        if raw_mode not in OPERATIONS:
+            _LOGGER.warning("Mode %s not found in known modes %s",
+                            raw_mode, OPERATIONS)
+        self._current_operation = OPERATIONS[raw_mode]
+
+        raw_temp = result.get('occupied_heating_setpoint')
+        if raw_temp:
+            self._target_temperature = raw_temp / 100.0
+        raw_temp = result.get('local_temp')
+        if raw_temp:
+            self._current_temperature = raw_temp / 100.0
+
+    @property
+    def supported_features(self):
+        """Return the list of supported features."""
+        return SUPPORT_FLAGS
+
+    @property
+    def temperature_unit(self):
+        """Return the unit of measurement."""
+        return TEMP_CELSIUS
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        return self._current_temperature
+
+    @property
+    def current_operation(self):
+        """Return the current operation mode."""
+        return self._current_operation
+
+    @property
+    def operation_list(self):
+        """Return a list of available operation modes."""
+        return OPERATION_LIST
+
+    @property
+    def target_temperature(self):
+        """Return the temperature we try to reach."""
+        if self._target_temperature is None or self._target_temperature < 0:
+            return None
+        return self._target_temperature
+
+    @asyncio.coroutine
+    def async_set_temperature(self, **kwargs):
+        """Set new target temperature."""
+        operation_mode = kwargs.get(ATTR_OPERATION_MODE)
+        if operation_mode is not None:
+            yield from self.async_set_operation_mode(operation_mode)
+
+        temperature = kwargs.get(ATTR_TEMPERATURE)
+        if temperature is None:
+            return
+        yield from self._wait_for_clean_communication()
+
+        temperature = temperature * 100
+        data = {'occupied_heating_setpoint': temperature}
+        res = yield from self._endpoint.thermostat.write_attributes(data)
+        self.last_request = time.time()
+        # Check result
+        for result in res:
+            for subresult in result:
+                if subresult.status != NO_ERROR_ANSWER_CODE:
+                    _LOGGER.error("Error setting operation mode: %s",
```

Comment:
Is this message correct? Seems you're showing operation mode message for result of temperature set.